pkg/syncer/translator/from_config_translator.go

```diff
@@ -0,0 +1,181 @@
+		gvk:            gvk,
+		eventRecorder:  ctx.VirtualManager.GetEventRecorderFor("from-host-" + strings.ToLower(gvk.Kind) + "-syncer"),
+		virtualToHost:  virtualToHost,
+		hostToVirtual:  hostToVirtual,
+		namespace:      ctx.Config.ControlPlaneNamespace,
+		translatorName: "from-host-" + strings.ToLower(gvk.Kind),
+		skipFuncs:      skipFuncs,
+	}, nil
+}
+
+func (c *fromHostTranslate) Name() string {
+	return c.translatorName
+}
+
+func (c *fromHostTranslate) Resource() client.Object {
+	switch c.gvk.Kind {
+	case "ConfigMap":
+		return &corev1.ConfigMap{}
+	case "Secret":
+		return &corev1.Secret{}
+	default:
+		return nil
+	}
+}
+
+func (c *fromHostTranslate) Migrate(_ *synccontext.RegisterContext, _ synccontext.Mapper) error {
+	return nil
+}
+
+func (c *fromHostTranslate) GroupVersionKind() schema.GroupVersionKind {
+	return c.gvk
+}
+
+func (c *fromHostTranslate) VirtualToHost(_ *synccontext.SyncContext, req types.NamespacedName, _ client.Object) types.NamespacedName {
+	vName, vNs := req.Name, req.Namespace
+	nn, _ := matchesVirtualObject(vNs, vName, c.virtualToHost, c.namespace)
+	return nn
+}
+
+func (c *fromHostTranslate) HostToVirtual(_ *synccontext.SyncContext, req types.NamespacedName, _ client.Object) types.NamespacedName {
+	nn, ok := matchesHostObject(req.Name, req.Namespace, c.hostToVirtual, c.namespace, c.skipFuncs...)
+	if !ok {
+		return types.NamespacedName{}
+	}
+	return nn
+}
+
+func (c *fromHostTranslate) IsManaged(_ *synccontext.SyncContext, pObj client.Object) (bool, error) {
+	hostName, hostNs := pObj.GetName(), pObj.GetNamespace()
+	_, managed := matchesHostObject(hostName, hostNs, c.hostToVirtual, c.namespace, c.skipFuncs...)
+	return managed, nil
+}
+
+func (c *fromHostTranslate) EventRecorder() record.EventRecorder {
+	return c.eventRecorder
+}
+
+func (c *fromHostTranslate) MatchesHostObject(hostName, hostNamespace string) (types.NamespacedName, bool) {
+	return matchesHostObject(hostName, hostNamespace, c.hostToVirtual, c.namespace, c.skipFuncs...)
+}
+func (c *fromHostTranslate) MatchesVirtualObject(virtualName, virtualNamespace string) (types.NamespacedName, bool) {
+	return matchesVirtualObject(virtualNamespace, virtualName, c.virtualToHost, c.namespace)
+}
+
+func matchesHostObject(hostName, hostNamespace string, resourceMappings map[string]string, vClusterHostNamespace string, skippers ...ShouldSkipHostObjectFunc) (types.NamespacedName, bool) {
+	for _, skipFunc := range skippers {
+		if skipFunc(hostName, hostNamespace) {
+			return types.NamespacedName{}, false
+		}
+	}
+
+	key := hostNamespace + "/" + hostName
+	matchesAllKeyInNamespaceKey := hostNamespace + "/*"
+
+	// first, let's try matching by namespace/name
+	if virtual, ok := resourceMappings[key]; ok {
+		virtualParts := strings.Split(virtual, "/")
+		if len(virtualParts) == 2 {
+			ns := virtualParts[0]
+			name := virtualParts[1]
+			if name == "*" {
+				name = hostName
+			}
+			return types.NamespacedName{Namespace: ns, Name: name}, true
```

Comment:
I just learned from Jan's [PR](https://github.com/loft-sh/loft-enterprise/pull/3706/files?w=1#diff-e7b4913cc69eb9882b7b87493cdd22c0b1286b0529a03aa26078c234c8abec2fR57) that `strings.Cut` exists. Its a nit suggestion, but this and other splits could be changed to something like. 
```suggestion
		ns, name, _ := strings.Cut(virtual, "/")
		if name == "*" {
			name = hostName
		}
		return types.NamespacedName{Namespace: ns, Name: name}, true
```

Reply:
good call :+1: